### PR TITLE
added ability to lookup projects by folders or project ids

### PR DESCRIPTION
**Pull request by alx13**

### CODE REVIEW

File: fast/stages/2-security/vpc-sc.tf [hunk 1]
```diff
@@ -19,38 +19,117 @@ locals {
...
+      from = v.from
+      to = merge([
+        v.to,
+        {
+          resources = sort(toset(flatten([
+            v.to.resources,
+            [for f in v.to.folder_ids : local._vpc_sc_resources_projects_in_folders[f]],
+            [for f in v.to.project_ids : local._vpc_sc_resources_projects_by_ids[f]],
+          ])))
+          project_ids = null
```
Comment (juliocc) [inline fast/stages/2-security/vpc-sc.tf:87]: I think this is ignored by the [provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/access_context_manager_service_perimeter). Looks lit it only accepts resources in the form of projects/number

Comment (alx13) [reply fast/stages/2-security/vpc-sc.tf:87]: Yes, I nulled it in case of underlying module changes

File: fast/stages/2-security/README.md [hunk 2]
```diff
@@ -14,21 +14,30 @@ The following diagram illustrates the high-level design of created resources and
...
+    - [Running the stage](#running-the-stage)
+  - [Customizations](#customizations)
+    - [KMS keys](#kms-keys)
+    - [VPC Service Controls configuration](#vpc-service-controls-configuration)
+      - [Dry-run vs. enforced](#dry-run-vs-enforced)
+      - [Access levels](#access-levels)
+      - [Lookup variables](#lookup-variables)
+      - [Ingress and Egress policies](#ingress-and-egress-policies)
+      - [Perimeters](#perimeters)
+    - [Brdiges](#brdiges)
```
Comment (ludoo) [inline fast/stages/2-security/README.md:36]: Typo (not yours probably but worth fixing)

File: fast/stages/2-security/README.md [hunk 3]
```diff
@@ -185,12 +194,13 @@ The script will create one keyring for each specified location and keys on each
 
 A set of variables allows configuring the VPC SC perimeters described above:
 
-- `vpc_sc_perimeter_projects` configures project membership in the three regular perimeters
+- `vpc_sc_perimeter_projects` configures project membership in regular perimeters
+- `vpc_sc_bridges` configures bridges between perimters
```
Comment (ludoo) [inline fast/stages/2-security/README.md:198]: typo

File: fast/stages/2-security/README.md [hunk 4]
```diff
@@ -258,24 +288,55 @@ Access levels and egress/ingress policies are referenced in perimeters via keys.
...
+    resources            = ["projects/0000000000"]
+    resources            = ["projects/2222222222"]
+    project_ids          = ["xxx-project-id"]
+    folder_ids           = ["555555555"]
+    folder_ids_recursive = ["666666666"]
   }
 }
 ```
 
+### Brdiges
```
Comment (ludoo) [inline fast/stages/2-security/README.md:319]: typo

**Review by ludoo — APPROVED** [review]

**Comment by ludoo** [comment]
@juliocc @drebes this will disrupt our standard VPC SC design, by making the stage more flexible. While I see the benefits, we will need to change all the supporting materials before a release is cut, including workflow slides.

**Comment by ludoo** [comment]
Won't the diagram and architecture also need to change, as the variables do not enforce a standard design anymore?

**Comment by juliocc** [comment]
I can't pinpoint a specific issue with this PR but I don't like the implementation. I **do like** the idea behind it but the code just looks messy to me. 

I particularly dislike the new `vpc-sc-resources.tf` but it could be that I can't follow what's  going on.

**Comment by ludoo** [comment]
Agree, can we move the code to its own module, which can be used to resolve ids to names?

**Comment by alx13** [comment]
`vpc-sc-resources.tf` is doing three things:

1. it puts all references to project by ids, folders to fetch projects inside recursive and non-recursive to the respective set, to deduplicate API lookups
2. creates lookup tables: `project id => project number`, `folder id => projects number` (recursive and not)
3. running lookups to form project lists for different inputs of VPC SC

**Comment by ludoo** [comment]
> Agree, can we move the code to its own module, which can be used to resolve ids to names?

**Comment by alx13** [comment]
I can, but let's get @juliocc onboard with this.

**Comment by ludoo** [comment]
> I can, but let's get @juliocc onboard with this.

we're usually very much aligned...

**Comment by rosmo** [comment]
I think the code looks fine, for what's it worth. "I don't like it" doesn't sound good.

**Comment by ludoo** [comment]
> I think the code looks fine, for what's it worth. "I don't like it" doesn't sound good.

it does instead, if the good code makes the whole stage harder to understand or modiify; anyway, I think Julio and I are aligned, let me close this and feel free to reopen as a standalone module, which I would very much like to have in Fabric...

**Pull request closed without merging**